sra2sal_byx.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+########################################################################
+#Script Name: sra2sal_byx.sh
+#Description: linker script to run experiment based on job array index
+#Args:        job array index
+#Author:      Georgia Doing
+#Date:        12-17-20
+#Email:       Georgia.Doing.GR@Dartmouth.edu
+########################################################################
+
+
+readarray SRXS < <(find sra_comp -mindepth 2 -maxdepth 2 -type d)
+./salPa_byx.sh ${SRXS[$1]}
```

Comment:
I'm not as familiar with job arrays, how does this work?

Reply:
It's specific to PBS scripting and is just a convenient way of submitting many jobs that are essentially the same but take different inputs or parameters. So here the job array passes an index to sra2sal_byx.sh, which then uses that index ($1) to select which SRP directory it is going to process. Here $1 will be an int ranging from 1-3000 since the length of SRXS is <3000 (there are < 3000 SRP projects with P.a.) - I just set 3000 as a safe upper limit. The job array is nice because it submits a small number of jobs at a time in order to abide by user limits. (I mostly learned about them from https://rc.dartmouth.edu/index.php/using-discovery/scheduling-jobs/using-job-arrays/ which may give a more clear explanation, haha)